ts/main.ts

```diff
@@ -1,64 +1,92 @@
-    return [];
   }
 }
 
-function renderAnime(animeList: Anime[]): void {
-  animeContainer.innerHTML = '';
+// REnders the anime cards
+function renderAnime(anime: Anime): HTMLDivElement {
+  const animeDiv = document.createElement('div');
+  animeDiv.classList.add('animes');
 
-  animeList.forEach((anime) => {
-    const animeDiv = document.createElement('div');
-    animeDiv.classList.add('animes');
+  const imageDiv = document.createElement('div');
+  imageDiv.classList.add('image-container');
+  const img = document.createElement('img');
+  img.src = anime.images.jpg.image_url;
+  img.alt = anime.title;
+  imageDiv.appendChild(img);
 
-    const imageDiv = document.createElement('div');
-    imageDiv.classList.add('image-container');
-    const img = document.createElement('img');
-    img.src = anime.images.jpg.image_url;
-    img.alt = anime.title;
-    imageDiv.appendChild(img);
+  const textDiv = document.createElement('div');
+  textDiv.classList.add('text-container');
+  const title = document.createElement('h2');
+  title.textContent = anime.title;
+  const episodes = document.createElement('p');
+  episodes.textContent = `Episodes: ${anime.episodes ?? 'N/A'}`;
 
-    const textDiv = document.createElement('div');
-    textDiv.classList.add('text-container');
-    const title = document.createElement('h2');
-    title.textContent = anime.title;
-    const episodes = document.createElement('p');
-    episodes.textContent = `Episodes: ${anime.episodes ?? 'N/A'}`;
+  textDiv.appendChild(title);
+  textDiv.appendChild(episodes);
 
-    textDiv.appendChild(title);
-    textDiv.appendChild(episodes);
+  // Add an event listener to the title to swap views
+  title.addEventListener('click', () => {
+    swapView(anime);
+  });
 
-    animeDiv.appendChild(imageDiv);
-    animeDiv.appendChild(textDiv);
+  animeDiv.appendChild(imageDiv);
+  animeDiv.appendChild(textDiv);
 
-    animeContainer.appendChild(animeDiv);
-  });
+  return animeDiv;
 }
 
-async function init(): Promise<void> {
-  const animeList = await fetchAnime();
-  renderAnime(animeList);
+// SWAPS THE VIEW . im pretty sure idk it wasnt working for some reason but now it is
+function swapView(anime: Anime): void {
+  const detailView = document.querySelector('.detail-view') as HTMLDivElement;
+  const titleElement = detailView.querySelector(
+    '.detail-title h1',
+  ) as HTMLHeadingElement;
+  const synopsisElement = detailView.querySelector(
+    '.detail-summary p',
+  ) as HTMLParagraphElement;
+  const imageElement = detailView.querySelector(
+    '.detail-image',
+  ) as HTMLDivElement;
+  const ratingElement = detailView.querySelector(
+    '.detail-rating h2',
+  ) as HTMLHeadingElement;
+
+  titleElement.textContent = anime.title;
+  synopsisElement.textContent = anime.synopsis;
+  imageElement.innerHTML = `<img src="${anime.images.jpg.image_url}" alt="${anime.title}">`;
+  ratingElement.textContent = `Rating: ${anime.rating ?? 'N/A'}`;
+
```

Comment:
**For your next feature:** No `innerHTML` allowed. `innerHTML` is a security issue. Even though you are setting it here, in theory a user could pass through the string of a script tag and run JavaScript code because of the `innerHTML`.